DTS-STN/Alpha-Site
# Unsubscribe

# Description

[Unsubscribe](https://trello.com/c/UCwRGy3G)

~~*This is still a work in progress*~~

An api endpoint will handle the form submission from /unsubscribe, which will generate a link and send it to the provided email address. Clicking the link will then perform the unsubscribe functionality, which is to remove all data from the record except the cuid.

## TODOs
Most of these can be done after we get actual designs, this PR can be looked at as just adding the functionality and the api endpoint, not the actual pages just yet.

- A confirmation page after a user unsubscribes
- Add a message on the /unsubscribe page after the user submits the form informing the user to check their email? Either that or have a second confirmation page - similar to how the signup/validate works
- ~~Unit testing~~
- E2E testing
- translations

## Acceptance Criteria

A user has the ability to unsubscribe from receiving invites to new experiments, or any other messages they might receive. There are two ways to do this: 
1. through a link provided in any email correspondence
2. the /unsubscribe page

After following the  /api/unsubscribe link from an email, the corresponding record in mongo contains only the cuid.

## Test Instructions

1. Follow the signup process, there is no need to do the validation
2. Navigate to the /unsubscribe page
3. Enter the email you used to sign up
4. Check the email with Unsubscribe in the subject and follow the provided link
5. Check your mongo database and see that the record is now only the cuid

## Checklist


## Product and Sprint Backlog

[Trello](https://trello.com/b/ZqWtJSyh/alpha-site-board)
